docs/source/optimize.rst

```diff
@@ -106,6 +106,7 @@ tasks to improve efficiency using the ``inline`` function. For example:
 .. code-block:: python
 
     >>> from dask.optimize import inline
+    >>> dependencies = {k: set(deps) for k, deps in dependencies.items()}
```

Comment:
Perhaps just add a check in `inline` instead?

```python
if dependencies is None:
    ...
elif dependencies and isinstance(dependencies.values()[0], list):
    dependencies = {k: set(v) for k, v in dependencies.items()}
```

Either is fine by me though.